controller/deployer/strategies/main.go

```diff
@@ -39,60 +141,87 @@ func jobEventsEqual(expected, actual jobEvents) bool {
 	return true
 }
 
-type jobEvents map[string]map[string]int
-
-func waitForJobEvents(events chan *ct.JobEvent, deployEvents chan<- ct.DeploymentEvent, releaseID string, expected jobEvents, log log15.Logger) error {
+func (d *Deploy) waitForJobEvents(releaseID string, expected jobEvents, log log15.Logger) error {
 	actual := make(jobEvents)
-outer:
+
+	handleEvent := func(typ, state string) {
+		if _, ok := actual[typ]; !ok {
+			actual[typ] = make(map[string]int)
+		}
+		actual[typ][state] += 1
+		d.deployEvents <- ct.DeploymentEvent{
```

Comment:
Should we dedup using the job ID? It seems like there are a variety of edge cases where we could get duplicate events.